posthog/api/capture.py

```diff
@@ -308,6 +326,16 @@ def get_event(request):
             if db_error:
                 send_events_to_dead_letter_queue = True
 
+    try:
+        invalid_token_reason = _check_token_shape(token)
+        if invalid_token_reason:
+            # TODO: check the capture_token_shape_invalid_total metric for false positives,
+            #  then move higher and refuse requests
+            TOKEN_SHAPE_INVALID_COUNTER.labels(reason=invalid_token_reason).inc()
+            logger.warning("capture_token_shape_invalid", token=token, reason=invalid_token_reason)
+    except Exception as e:
```

Comment:
Are there cases where this will raise? I guess if token isn't actually a string? Could we instead add a check for this case and handle accordingly with a 400 and remove this try/expect?

Reply:
Or I guess, in this we're trying to not actually change behaviour, so perhaps just include that as a `reason`

Reply:
Indeed, that's running that logic as a no-op for now and reporting its false-positives.

Hopefully the logic should not throw, but I went extra-extra-safe for now ^^